src/iceberg/meson.build

```diff
@@ -0,0 +1,139 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+iceberg_include_dir = include_directories('..')
+iceberg_sources = files(
+    'arrow_c_data_guard_internal.cc',
+    'catalog/in_memory_catalog.cc',
+    'expression/expression.cc',
+    'expression/literal.cc',
+    'file_reader.cc',
+    'file_writer.cc',
+    'inheritable_metadata.cc',
+    'json_internal.cc',
+    'manifest_entry.cc',
+    'manifest_list.cc',
+    'manifest_reader.cc',
+    'manifest_reader_internal.cc',
+    'manifest_writer.cc',
+    'metadata_columns.cc',
+    'name_mapping.cc',
+    'partition_field.cc',
+    'partition_spec.cc',
+    'schema.cc',
+    'schema_field.cc',
+    'schema_internal.cc',
+    'schema_util.cc',
+    'snapshot.cc',
+    'sort_field.cc',
+    'sort_order.cc',
+    'statistics_file.cc',
+    'table.cc',
+    'table_metadata.cc',
+    'table_scan.cc',
+    'transform.cc',
+    'transform_function.cc',
+    'type.cc',
+    'util/decimal.cc',
+    'util/gzip_internal.cc',
+    'util/murmurhash3_internal.cc',
+    'util/timepoint.cc',
+)
+
+nanoarrow_dep = dependency('nanoarrow')
+nlohmann_json_dep = dependency('nlohmann_json')
+spdlog_dep = dependency('spdlog')
+zlib_dep = dependency('zlib')
+
+iceberg_deps = [nanoarrow_dep, nlohmann_json_dep, spdlog_dep, zlib_dep]
+
+iceberg_lib = library(
```

Comment:
Any plan to add `iceberg_bundle_lib`?

Reply:
No, I mean this specific library: https://github.com/apache/iceberg-cpp/blob/main/src/iceberg/CMakeLists.txt#L164. Perhaps the name is misleading. It provides builtin support for Parquet, FileSystem and Avro implementations.